common/oatbox/log/StreamHandler.php

```diff
@@ -0,0 +1,75 @@
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\oatbox\log;
+
+use Monolog\Handler\StreamHandler as MonologStreamHandler;
+use Monolog\Logger;
+
+/**
+ * Stores to any stream resource
+ *
+ * Provides the ability to use command line parameters --log-file-path and --log-level:
+ *
+ * if the --log-file-path parameter is specified, the path to the file for logging is taken from this parameter,
+ * if this parameter is not specified, the path to the file is taken from the system configuration
+ * for example:
+ * php index.php 'oat\taoTaskQueue\scripts\tools\RunWorker' --log-file-path /var/www/html/data/tao/log/tao-nccer.log
+ *
+ * if the --log-level parameter is specified, the path to the file for logging is taken from this parameter,
+ * if this parameter is not specified, the path to the file is taken from the system configuration
+ * for example:
+ * php index.php 'oat\taoTaskQueue\scripts\tools\RunWorker' --log-level 100
+ *
+ * @author Andrey Niahrou, <andrei.niahrou@taotesting.com>
+ */
+class StreamHandler extends MonologStreamHandler
+{
+    /**
+     * @param resource|string $defaultStream  (Unless this parameter is specified on the command line as --log-file-path, otherwise it is ignored) resource where data will be output
+     * @param int             $defaultLevel   (Unless this parameter is specified on the command line as --log-level, otherwise it is ignored) The minimum logging level at which this handler will be triggered
+     * @param bool            $bubble         Whether the messages that are handled can bubble up the stack or not
+     * @param int|null        $filePermission Optional file permissions (default (0644) are only for owner read/write)
+     * @param bool            $useLocking     Try to lock log file before doing any writes
+     *
+     * @throws \Exception                If a missing directory is not buildable
+     * @throws \InvalidArgumentException If stream is not a resource or string
+     */
+    public function __construct($defaultStream, int $defaultLevel = Logger::DEBUG, bool $bubble = true, $filePermission = null, bool $useLocking = false)
+    {
+        $stream = $this->getScriptParameter('--log-file-path') ?: $defaultStream;
```

Comment:
```suggestion
        $stream = $this->getScriptParameter('--log-file') ?: $defaultStream;
```
Path in this case not needed, better to explain it in the description.